Pinned doc deps

MichaelHatherly opened the pull request:
As requested by @tkelman, pinned versions of documentation dependencies. Also reinstates the stacktraces stdlib file that had been left out of the index.

MichaelHatherly committed:
Reinstate stacktraces stdlib file

Also removes C.md and julia.md index files from devdocs since their
content is a duplicate of what is found in the nav menu.

MichaelHatherly committed:
Pin package versions for doc deps

Versions must be kept exact to allow for reproducible doc builds.

MichaelHatherly commented on doc/REQUIRE:
```diff
@@ -0,0 +1,3 @@
+Compat 0.9.5 0.9.5+
+DocStringExtensions 0.3.0 0.3.0+
+Documenter 0.8.2 0.8.2+
```
@tkelman, I assume you did mean pinning *all* the dependencies rather than just Documenter?

tkelman replied:
right

tkelman reviewed (approved):
> their content is a duplicate of what is found in the nav menu.

How does that nav menu get populated with an index then?

MichaelHatherly commented:
> How does that nav menu get populated with an index then?

Those two index pages contain the same set of page links as the two subsections in https://github.com/JuliaLang/julia/blob/4dcada105716dc30c8e7edcc843c7d8dcc9bffac/doc/make.jl#L87-L110. (`PAGES` is what is used to build the nav menu.)

MichaelHatherly committed:
Fix typo in page title

MichaelHatherly committed:
Make devdocs/ast.md formatting more uniform

Automatic conversion of RST definition lists into markdown left quite
a bit of room for improvement. This reformats all fieldname lists to
use a single layout, namely:

    `fieldname`

    Description of `fieldname`...

Also fixes up some sublists that were not correctly rendered in the
RST docs originally.

MichaelHatherly commented:
Also includes fixes for https://github.com/JuliaLang/julia/pull/18588#commitcomment-20126414 and https://github.com/JuliaLang/julia/pull/18588#discussion_r91622605.

The last commit reformats `devdocs/ast.md` quite a bit to get a more consistent layout, which I think is more readable and consistent.
